test/let_value_with_stop_token_test.cpp

```diff
@@ -0,0 +1,241 @@
+  }
+
+  inplace_stop_source& source_;
+};
+
+TEST(LetWithStopToken, Simple) {
+  timed_single_thread_context context;
+
+  // Simple usage of 'let_value_with_stop_token()'
+  // - Sets up some work to execute when receiver is cancelled
+  int external_context = 0;
+  optional<int> result =
+      sync_wait(let_value_with_stop_source([&](auto& stopSource) {
+        return let_value_with_stop_token([&](inplace_stop_token stopToken) {
+          // Needs to pass the stop token by value into the capture list
+          // to prevent accessing stopToken reference after function has
+          // returned.
+          return let_value_with(
+              [stopToken, &external_context]() noexcept {
+                auto stopCallback = [&]() mutable noexcept {
+                  external_context = 42;
+                };
+                using stop_callback_t =
+                    typename inplace_stop_token::template callback_type<
+                        decltype(stopCallback)>;
+                return stop_callback_t{stopToken, stopCallback};
+              },
+              [&stopSource](auto&) -> unifex::any_sender_of<int> {
+                stopSource.request_stop();
+                return just_done();
+              });
+        });
+      }));
+
+  EXPECT_TRUE(!result);
+  EXPECT_EQ(external_context, 42);
+}
+
+TEST(LetWithStopToken, SimpleInplaceStoppableNoexcept) {
+  static_assert(std::is_same_v<
+                stop_token_type_t<InplaceStoppableIntReceiver>,
+                inplace_stop_token>);
+
+  // Simple usage of 'let_value_with_stop_token()' with receiver with inplace
+  // stop token
+  // - Sets up some work to execute when receiver is cancelled
+  int external_context = 0;
+  inplace_stop_source stopSource;
+  auto stop_token_functor = [&](inplace_stop_token stopToken) noexcept {
+    return let_value_with(
+        [stopToken, &external_context]() noexcept {
+          auto stopCallback = [&]() noexcept {
+            external_context = 42;
+          };
+          using stop_callback_t =
+                    typename inplace_stop_token::template callback_type<
+                        decltype(stopCallback)>;
+          return stop_callback_t{stopToken, stopCallback};
+        },
+        [&](auto&) noexcept {
+          stopSource.request_stop();
+          return just_done();
+        });
+  };
+  static_assert(is_nothrow_connectable_v<
+                callable_result_t<
+                    decltype(let_value_with_stop_token),
+                    decltype(stop_token_functor)>,
+                InplaceStoppableIntReceiver>);
+  auto op = unifex::connect(
+      let_value_with_stop_token(std::move(stop_token_functor)),
+      InplaceStoppableIntReceiver{stopSource});
+  unifex::start(op);
+
+  EXPECT_EQ(external_context, 42);
+}
+
+TEST(LetWithStopToken, SimpleUnstoppable) {
+  static_assert(is_stop_never_possible_v<
+                stop_token_type_t<UnstoppableSimpleIntReceiver>>);
```

Comment:
Huh.  This `static_assert` is failing when building with MSVC 2019 in C++20 mode.  Weird.